### PR TITLE
DNM: Bpfd integration

### PR DESCRIPTION
**Pull request by astoycos**
This is a super rough first draft at integrating bpfd with kepler, it is meant to house some early explorations.  It's design will change dramatically before there's actually something we can merge since bpfd is currently undergoing some large scale design changes which will make the integration even easier.

Specifically moving forward maps will not be shared via a hostpath volume ultimately opening the door for the kepler exporter to be unprivileged when interacting with it's ebpfProgram maps.

### CODE REVIEW

**Comment by rootfs** [comment]
@astoycos this is cool! 
 
can you make it a kustomize patch (you'll find some [here](https://github.com/sustainable-computing-io/kepler/tree/main/manifests/config/exporter/patch)) and add an [option](https://github.com/sustainable-computing-io/kepler/tree/main/manifests) to it?

File: pkg/bpfassets/attacher/libbpf_attacher.go [hunk 1]
```diff
@@ -101,40 +105,95 @@ func attachLibbpfModule() (*bpf.Module, error) {
...
-		_, err = irq_prog.AttachTracepoint("irq", "softirq_entry")
-		if err != nil {
-			klog.Infof("failed to attach irq/softirq_entry: %v", err)
-			// disable IRQ metric
-			config.ExposeIRQCounterMetrics = false
-		}
+		return nil, fmt.Errorf("Can't get map")
+	}
+	klog.Infof("setting pin path for map %s", PidTimeMapName)
+	if err := pidTimeMap.SetPinPath(path.Join("/run/bpfd/fs/maps/5ef54022-f0a7-42d1-afd7-4868686edff6", PidTimeMapName)); err != nil {
```
Comment (astoycos) [inline pkg/bpfassets/attacher/libbpf_attacher.go:132]: This will not be hardcoded forever : -)

**Comment by astoycos** [comment]
Thanks @rootfs and most definitely this is just some early hacking, I don't think it's even quite ready for a review yet :) 

At the very least for the flow to work properly I'll need to explore deprecating libbpf + bcc in favor of using clilium's ebpf/go

File: pkg/bpfassets/attacher/libbpf_attacher.go [hunk 2]
```diff
@@ -295,13 +354,20 @@ func resizeArrayEntries(name string, size int) error {
...
-		return err
+	klog.Infof("setting pin path for map %s", name)
+	if err := m.SetPinPath(path.Join("/run/bpfd/fs/maps/5ef54022-f0a7-42d1-afd7-4868686edff6", name)); err != nil {
+		return fmt.Errorf("error pinning map %s: %w", name, err)
 	}
 
-	if current := m.GetMaxEntries(); current != uint32(size) {
-		return fmt.Errorf("failed to resize map %s, expected %d, returned %d", name, size, current)
-	}
+	// BPFD HACK don't resize for now or else libbpf can't reuse pins
```
Comment (astoycos) [inline pkg/bpfassets/attacher/libbpf_attacher.go:362]: This will be eventually be taken care of in the bpfd YAML

**Comment by vimalk78** [comment]
Awesome, this should simplify things. 
@astoycos can you please link bpfd documentation so that we can follow along

**Comment by astoycos** [comment]
Sure @vimalk78 PTAL at [bpfd.dev](https://bpfd.dev/)

**Pull request closed without merging**